### ISSUE TITLE
Linux 20.04 - deno install not specifying path in bash

### ISSUE THREAD

**Issue by HenrySabio** [denoland/deno#8241]
After some trial and error I realized that I needed to manually add the following to my `.bashrc` file:

```sh
export DENO_INSTALL="/home/$USER/.deno"
export PATH="$DENO_INSTALL/bin:$PATH"
```
Otherwise `deno` command would return an error as if not installed unless I used the direct path instead.
This should either be added to the `readme` file under installation section for MacOS/Linux or done for the user during install.

My system/environment:
Dell XPS - Developer Edition (Linux OEM)
Ubuntu 20.04 - Fresh Install, all default settings & apps out of box

(Contribution Newbie - forgive me if I missed references to this in existing Issues/PRs, improper formatting, or if this is intentional by design.)

**Comment by nayeemrmn** [comment]
https://github.com/denoland/deno_install/blob/4d960274a86fa46265ca5ba9adc01a2405baded1/install.sh#L62-L64 -- are you not seeing this?

**Comment by HenrySabio** [comment]
> https://github.com/denoland/deno_install/blob/4d960274a86fa46265ca5ba9adc01a2405baded1/install.sh#L62-L64 -- are you not seeing this?

No, the following was the only output after running the provided command to install:

```
user@pcname:~$ curl -fsSL https://deno.land/x/install/install.sh | sh
######################################################################## 100.0%
######################################################################### 100.0%
Archive:  /home/user/.deno/bin/deno.zip
  inflating: /home/user/.deno/bin/deno  
Deno was installed successfully to /home/user/.deno/bin/deno
Run 'deno --help' to get started
```

**Comment by nayeemrmn** [comment]
That would mean there is a `deno` in your PATH. Have you confirmed that `deno --help` doesn't work after seeing that output?

**Comment by HenrySabio** [comment]
Yes, as stated before, without manually adding it to my `.bashrc` file myself - typing `deno` returns the usual `Command 'deno' not found...` message for whenever a command is not valid.

For me to use deno (before adding manually to my `.bashrc` file) I would have to type `/home/user/.deno/bin/deno` instead.
If I trust my understanding of the scripts so far - I believe the issue is coming from `line 60` of the `deno_install/install.sh` file that you linked to.

```
shell_profile=".bash_profile"
```

There is no `.bash-profile` file by default when using bash - so it's checking for a condition that most people won't pass.

_In my experience_ - the `.bash_profile` file does not exist by default on most if not all systems.
`.bashrc` is always the default file until other files like `.bash_profile` or `.bash_alias` are created _manually_ by the user.

**Comment by nayeemrmn** [comment]
> I believe the issue is coming from `line 60` of the `deno_install/install.sh` file that you linked to.
> 
> ```
> shell_profile=".bash_profile"
> ```

No, that's just deciding what file to mention in the text it shows you. The relevant condition is https://github.com/denoland/deno_install/blob/4d960274a86fa46265ca5ba9adc01a2405baded1/install.sh#L55. What does `command -v deno; echo $?` output?

**Comment by HenrySabio** [comment]
If I enter it _before_ manually adding the path to my `.bashrc` - 1
If I enter it _after_ manually adding the path to my `.bashrc` - 0

**Comment by nayeemrmn** [comment]
Sorry for this, could you remove it from your `.bashrc` and run the `curl ... | sh` command one more time? This doesn't make a lot of sense.

**Comment by HenrySabio** [comment]
I understand.

I've run it 5 times with the same results.
Even after removing deno itself to do a fresh install and on a new account with no packages installed.

I get the same result everytime:
```user@pcname:~$ curl -fsSL https://deno.land/x/install/install.sh | sh
######################################################################## 100.0%
######################################################################### 100.0%
Archive:  /home/user/.deno/bin/deno.zip
  inflating: /home/user/.deno/bin/deno  
Deno was installed successfully to /home/user/.deno/bin/deno
Run 'deno --help' to get started
```

**Comment by bekalshenoy** [comment]
i am also facing this problem and i am also using same os

**Comment by MarkTiedemann** [comment]
@HenrySabio Did you `source`  your `.bashrc`?

Might be a dup of #6199 (which may be fixed by https://github.com/denoland/deno_install/pull/149).

**Comment by HenrySabio** [comment]
This would not solve this issue. 

I only have 1 `.bashrc` file. 
So reloading the file would just reload the same file. 

The problem is not my file. 
The problem is that the installer did not add the path to deno to my `.bashrc` file. Nor was there any direction to do so. 

I just happened to know that this was something that should have been done and applied the fix manually myself. 

Because of this I think there should be a disclaimer added to the install read me to let people know if the `deno` command does not work after install & reload they need to define the path in their `.bashrc` or appropriate file manually.

**Comment by nayeemrmn** [comment]
> Because of this I think there should be a disclaimer added to the install read me to let people know if the `deno` command does not work after install & reload they need to define the path in their `.bashrc` or appropriate file manually.

As I linked above, there are instructions printed from the installer script. For whatever reason it's not printed for you but that's a bug (which I can't reproduce).

**Comment by nayeemrmn** [comment]
Also this should be moved to https://github.com/denoland/deno_install.

**Comment by MarkTiedemann** [comment]
> This would not solve this issue. I only have 1 `.bashrc` file. So reloading the file would just reload the same file.

Whenever you edit your `.bash_profile`, you need to `source` it again.

> The problem is that the installer did not add the path to deno to my `.bashrc` file.

This is by design. The installer should not mess with people's config files.
